opendatahub-io/ai-edge
# Update image reference in edge Deployment

## Description


Adds another pipeline that changes the image reference in the `acm/odh-edge/apps/.../kustomization.yaml` file, pushes a new branch with the change to the on-prem Git clone of the `opendatahub/ai-edge` repo, and creates a PR with the changes.

## How Has This Been Tested?




Pipelines tested on the `pplaczek-pipeline-dev` and `gryan-pipelines` namespace on `local-cluster`.
Git tested on `http://gitea-ai-edge-rhoai-edge-acm.apps.rhoai-edge-acm-poc.rhaiseng.com/edge-user-1/ai-edge-gitops`.

See [reply below](https://github.com/opendatahub-io/ai-edge/pull/54#issuecomment-1703167501) for example screenshots.

### Testing instructions
If you would like to try this out, the following steps should be runnable in an isolation on the cluster:
- Create a project with `oc new-project <something>`
- Apply the required resources, with `oc apply -k ./pipelines/tekton/gitops-update-pipeline/`
- You can then either create a PipelineRun for the pipeline through the OpenShift Pipelines UI in the cluster (or presumably using `tkn` CLI), or by using one (or both) of the pre-canned manifests, like so:
```
oc create -f ./pipelines/tekton/gitops-update-pipeline/gitops-update-pipelinerun-bike-rentals.yaml
oc create -f ./pipelines/tekton/gitops-update-pipeline/gitops-update-pipelinerun-tensorflow-housing.yaml
```
- You should then see the pipeline running immediately through the web console, and you should see a PR created in the relevant repo at the end. For the bike-rentals one that will be [here](http://gitea-ai-edge-rhoai-edge-acm.apps.rhoai-edge-acm-poc.rhaiseng.com/edge-user-1/ai-edge-gitops/pulls), and for the tensorflow-housing one it will be [here](http://gitea-ai-edge-rhoai-edge-acm.apps.rhoai-edge-acm-poc.rhaiseng.com/edge-user-2/ai-edge-gitops/pulls). 

## Merge criteria:
